lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -152,7 +178,7 @@ def test_put_reserved_metadata(self, client, server_config, attach_dataset):
         )
         assert response.status_code == HTTPStatus.BAD_REQUEST
         assert response.json == {
-            "message": "Unrecognized JSON keys access given for parameter metadata; allowed keywords are saved,seen,user"
+            "message": "Unrecognized JSON keys access given for parameter metadata; allowed keywords are dashboard.saved,dashboard.seen,user."
```

Comment:
Worrying about the number of the word "key(s)" seemed silly for the exception text; however, yeah, as we're passing that on through the API I likely should have worried about it. That gets complicated when we eventually localize messages, though, as number is difficult to change (you really need separate message templates). Maybe "one or more unrecognized JSON key"?? (Ugh.) And the `access` is a `",".join(list)` so I can't just quote it, I'd need to add code to quote each keyword. (Though I suppose I could just throw the `repr` of each on the list.) It didn't seem justified at the time, but maybe I was wrong. (Sigh.)

Reply:
Actually, using "key(s)" rather than actively adding or omitting the pluralization would be fine.

I'm more concerned with making it clear that the key (e.g., `access` in the example, here) is a "token" and not part of the (here) English language of the message.  (The text of the message above is hard for me as a human to parse without the specialized knowledge that "access" is an abstract noun and not some other part of the sentence.)

If you modified the message text to put a comma before the `",".join(list)` and a comma after it, that would suffice.  It would also suffice to put the entire, joined list inside a single pair of double quotes.  I just want something to set it off from the surrounding English language structure.  (But, if you really want to put each key into double quotes for me, I'm sure I can come up with some nice Python to do that.... 😉)

Reply:
Yeah, I thought of just using "key(s)" but I never liked that cheat. That might be the most pragmatic solution when we eventually move to a message catalog, (vs having `BAD_KEY_MESSAGE_1` and `BAD_KEY_MESSAGE_2`; but for now I just added the "s" if there's more than one.

Reply:
Ha!  No sooner did I hit the submit button than I came up with a nice approach.  (My brain is definitely an asynchronous, data-flow computing device.)  How about this:
```
    def __str__(self):
        return 'Unrecognized {} "{}" given for parameter {}; allowed keywords are "{}"'.format(
            self.expected_type,
            '", "'.join(self.unrecognized),
            self.parameter.name,
            '", "'.join(self.keywords)
        )
```